### DISCUSSION TITLE
globe view experiments

### DISCUSSION THREAD

**Discussion by kikuchan** [openlayers/openlayers#16109] [Show and tell]
![image](https://github.com/user-attachments/assets/233ec7dd-d3e5-4c3b-b904-ff6f9c409dbd)

This is a cheat by post-processing on framebuffer for now. haha.

Although it cannot be overlaid with 3D objects, or used for extruded terrain, is it worth continuing...? 🤔

**Comment by jahow** [comment]
Hi @kikuchan! Impressive effort. It would be interesting to see the code you used to achieve this!

One thing that I find interesting is that having an API to do rendering pipelines would probably be very useful in a case like that. https://github.com/openlayers/openlayers/pull/14491 is another example where a good rendering pipeline system would be very relevant.

Then people could come up with whatever tricks and adjustments that would fit their needs, including showing the map as a globe instead :slightly_smiling_face:

**Comment by kikuchan** [comment]
Hi, thank you for your message and information!

Here's the code:
https://github.com/kikuchan/openlayers/blob/experiment-globe-view/examples/webgl-globe-view.js
It almost works, but there are glitches. 😉 

And, I had to refactor PostProcessingPass.
https://github.com/kikuchan/openlayers/commit/3c82bbc432a2f022a73ed939ffa6d0971c5647ab
removes duplicated codes, and moves it to Helper class.

**Reply by jahow** [reply]
Thank you for sharing this, it is really interesting. I wish we could offer some kind of post processing shader builder to avoid having to write all the shaders. I think it's interesting that the helper manages the post processes too.

Basically a good target for better post processing system would be:
* have it work for a heatmap layer
* have it work for a particle flow layer
* have it work for your globe experiment

**Comment by kikuchan** [comment]
I'm still working on it. tilt and perspective.
![image](https://github.com/user-attachments/assets/f6465de2-30d3-4863-a3e2-e013cf7bb74e)
